users/views.py

```diff
@@ -131,6 +131,13 @@ def get(self, request, *args, **kwargs):
                 with translation.override(locale):
                     return super().get(request, *args, **kwargs)
 
+        # Save the client_id to the session to be used in the HelsinkiTunnistus
+        # social auth backend.
+        request.session["oidc_authorize_original_client_id"] = request.GET.get(
```

Comment:
Validation: this is using a query param as-is. I do imagine the actual client id is validated later, but it's kind of implicit. I suppose it might be difficult to only store this value after validation?

Reply:
Yes it will be validated in the parent get method, but I can of course check it here too.